### ISSUE TITLE
Support for [APP Powered by Tuya] DIGOO DG-HAMB 2G Alarm

### ISSUE THREAD

**Issue by Asom-Velz** [rospogrigio/localtuya#192]
Hi 

do you plan to support for the DIGOO DG-HAMB 2G alarm which is working with APP Powered by Tuya?
![IMG_9221](https://user-images.githubusercontent.com/65943619/100375721-e8ea1200-3016-11eb-98b3-e24c6e04031c.jpg)
![IMG_9222](https://user-images.githubusercontent.com/65943619/100375726-ea1b3f00-3016-11eb-9b5a-f4b6ba7495a3.jpg)
![IMG_9223](https://user-images.githubusercontent.com/65943619/100375730-eab3d580-3016-11eb-9669-6ef5e9eb276c.jpg)
![IMG_9224](https://user-images.githubusercontent.com/65943619/100375731-eb4c6c00-3016-11eb-8274-a6f13ec0cd01.jpg)

**Comment by postlund** [comment]
Probably at some point. We don't really have a roadmap or general plan for what we add, it's driven by requests and if someone wants to implement it. Would imagine this is similar to #119.

To help out, please attach a DPS dump of your device.

https://github.com/rospogrigio/localtuya/wiki/HOWTO-get-a-DPs-dump

**Comment by Shaman2** [comment]
> 
> 
> Probably at some point. We don't really have a roadmap or general plan for what we add, it's driven by requests and if someone wants to implement it. Would imagine this is similar to #119.
> 
> To help out, please attach a DPS dump of your device.
> 
> https://github.com/rospogrigio/localtuya/wiki/HOWTO-get-a-DPs-dump

yes, may it the some as 119

**Comment by postlund** [comment]
Likely yes, but it probably uses other datapoints so dump would be great!

**Comment by Asom-Velz** [comment]
Than you for your replies,

I read the article of how to create a DPS dump of my device but I have a question

Where I have to extract in order to run the test.py? 

I am running Home Assistant on a NUC Win10Pro + VM Workstation 16 
The only available location is the console in the VM Machine 

![HAonVM-rsz](https://user-images.githubusercontent.com/65943619/100469661-5a8d9300-30df-11eb-9696-ebba7f0cf89e.jpg)

Regards
Anastasios

**Comment by dademo-1999** [comment]
> Likely yes, but it probably uses other datapoints so dump would be great!

Hello, my device looks like being similar to [#119](https://github.com/rospogrigio/localtuya/issues/119) too. It's a **GAUTONE PG107 Wifi GSM Alarm** (I think it's manufactered by [PGST](http://www.cn-pgst.com/Products/Smart_Home_Alarm_System/3G___GSM_Alarm_System/130.html)).
I run DPS dump and that's the result:

  DPS [1] VALUE [disarmed] 
  DPS [2] VALUE [40] 
  DPS [3] VALUE [3] 
  DPS [4] VALUE [True] 
  DPS [6] VALUE [True] 
  DPS [12] VALUE [False] 
  DPS [13] VALUE [False] 
  DPS [20] VALUE [False] 
  DPS [21] VALUE [False] 
  DPS [27] VALUE [True] 
  DPS [28] VALUE [30] 
  DPS [101] VALUE [normal] 

Hope it helps.

**Comment by Asom-Velz** [comment]
Yes it seems that it is exactly the same model with different brand.

Still it would be grade if someone can give me a roadmap how to take a DPS dump since I am running on a NUC Win10Pro + VM Workstation 16 and I dont know were to run the  run the test.py script.

**Comment by dademo-1999** [comment]
> Yes it seems that it is exactly the same model with different brand.
> 
> Still it would be grade if someone can give me a roadmap how to take a DPS dump since I am running on a NUC Win10Pro + VM Workstation 16 and I dont know were to run the run the test.py script.

I run test.py from a linux machine (in my case was a Paralles Desktop VM machine). You don't need to run the script from the same Home Assistant machine. You just need to be on the same network and type the correct deviceID, IPaddress
and  localKey

**Comment by Asom-Velz** [comment]
Hi 
I don't have linux machine and also I don't have knowledge from linux. Any other solution based on windows machine or directly @  VM Machine console? 

![Clipboard-1](https://user-images.githubusercontent.com/65943619/105988459-50205300-60a8-11eb-9606-6046fc367020.jpg)

**Comment by dademo-1999** [comment]
> Hi
> I don't have linux machine and also I don't have knowledge from linux. Any other solution based on windows machine or directly @ VM Machine console?
> 
> ![Clipboard-1](https://user-images.githubusercontent.com/65943619/105988459-50205300-60a8-11eb-9606-6046fc367020.jpg)

Can't help sorry. But, running a simply usb live versione of Linux, I think it's easier than running the script from the VM workstation or directly from Windows. However, il you google "run linux script on windows", you can find different tutorials. Not sure it works and sure I can post the link here for the board policy .

**Comment by ropmip** [comment]
I have this model and I am able to integrate in homeassistant with this plug-in. Added all the dps as sensors to monitor changes. Only conclusion for now is status (dps1-arm/disarmed) and power status (dps101 - normal / powerDisconnect). Even with reduced funcionality this allows to use the alarm status as condition for automations.

**Comment by dademo-1999** [comment]
> I have this model and I am able to integrate in homeassistant with this plug-in. Added all the dps as sensors to monitor changes. Only conclusion for now is status (dps1-arm/disarmed) and power status (dps101 - normal / powerDisconnect). Even with reduced funcionality this allows to use the alarm status as condition for automations.

Have you find a way to arm/disarm through home assistant?

**Comment by ropmip** [comment]
> Have you find a way to arm/disarm through home assistant?
Yes! Just checked that I can use the service localtuya.set_dp to arm / disarm (dp: 1)
For digoo values for dp: 1 can be:
arm
disarmed
home
I guess this way it sould be possible to have a manual alarm on homeassistant (with it's own code) that is linked with the status on my tuya alarm.

**Comment by ropmip** [comment]
Full DPs from digoo (monitored from within the integrations- only saw changes on 1 and 101)
```
 DPS [1] VALUE [disarmed] / [arm] / [home]
 DPS [2] VALUE [40]
 DPS [3] VALUE [3]
 DPS [4] VALUE [True]
 DPS [6] VALUE [True]
 DPS [12] VALUE [False]
 DPS [13] VALUE [False]
 DPS [20] VALUE [True]
 DPS [21] VALUE [False]
 DPS [28] VALUE [30]
 DPS [101] VALUE [normal] / [powerDisconnect]
```

**Comment by Murada99** [comment]
Hi, I have the same alarm PG107, I got the localKey but if I setup the alarm with localtuya, the alarm is deleted from the smartlife application. Each time I try to pair again with the mobile application, the localKey is changed. 
I have to choice between localtuya or the mobile application.  Do you have the same problem ?

**Comment by ropmip** [comment]
No, is working fine in homeassistant and controlling in smartlife app at the same time.

**Comment by Murada99** [comment]
Hi ropmip, how do you get the localKey ?
I get it using the command: tuya-cli link --api-key xxxxx --api-secret xxxx --schema xxx .....
This command associates with the alarm but then I lost the alarm in the app.

**Comment by Murada99** [comment]
Oups, I get it working using another method.

**Comment by Asom-Velz** [comment]
> 
> 
> Oups, I get it working using another method.

Is it possible to tell me the way to get the local key?

**Comment by Murada99** [comment]
Yes, of course,
I did the method in this link 
https://github.com/codetheweb/tuyapi/blob/master/docs/SETUP.md

I started with "Linking a Tuya device with Smart Link" and but after the point 5, don't put your devices into linking mode, just do the  "Listing Tuya devices from the Tuya Smart or Smart Life apps" method.
Putting in linking mode, will generate a new local key.

**Comment by Murada99** [comment]
I'm trying now to set the alamr with the localtuya.sep_dp but without success:

service: localtuya.set_dp
data:
  device_id: xxxxxxx
  dp: 1
  value: arm

**Comment by Murada99** [comment]
I get this working.
Status arm / disarmed / home are available in HA and I can set the alarm from HA, thx ropmip.

The next step is to be alerted in HA when the alarm is triggered. Has anyone succeeded in this?

**Comment by ropmip** [comment]
To make it simple on HA side I added a "manual alarm" to the HA configuration and linked it's status to the tuya alarm (with automations). This way I have a lovelace alarm control panel in HA that always reflects the tuya alarm status and I can even control it from lovelace (with a custom PIN).
Full support implementation on the localtuya side could be to configure it directly as an alarm without the need for automations!

**Comment by Murada99** [comment]
That's a good solution to wotk with the manual alarm control panel.
How you get the information that the alarm was triggered ?
I don't see a DPS with this information.

**Comment by ropmip** [comment]
I guess you have no dps with that information. One workaround is to configure an external rf siren on the alarm and have an HA sensor to capture that rf signal. One other option is to capture rf signal on HA directly from motion detectors. I have a custom rflink bridge and already use those sensors for other automations on HA.

**Comment by Murada99** [comment]
thanks ! That sounds good, I have a rfxcom and an external siren, just need to capture the rf signal. Perfect solution.

**Comment by agarbato** [comment]
Thanks for sharing all the precious information guys :-)

I got mine finally working on HA.
It's a [TecPeak](https://www.amazon.it/gp/product/B08DCJZJPN/ref=ppx_yo_dt_b_asin_title_o06_s00?ie=UTF8&psc=1) alarm working with Tuya/Smartlife apps.

Most of the dps are the same.
Once I defined all the sensors changing settings on the app I was able to identify the correct things.

I share here my configuration.yaml , maybe it could be useful to somebody.
I guess there is no way to get sensors of pirs and other accessories since tuya-cli don't detect any.

I still don't understand how to have an alarm switch on lovelace.
I can only have sensors working but nothing else, I'm not HA expert though.

@ropmip you mention you use [alarm manual](https://www.home-assistant.io/integrations/manual/)
Can you share if you can some automation you did?

```localtuya:
  - host: 1.2.3.4
    device_id: xxxxxxxxxx
    local_key: xxxxxxxxxx
    friendly_name: Alarm
    protocol_version: "3.3"
    entities:
      - platform: sensor
        friendly_name: alarm.status
        id: 1
      - platform: sensor
        friendly_name: alarm.power
        id: 101
      - platform: sensor
        friendly_name: alarm.32
        id: 32
      - platform: sensor
        friendly_name: alarm.entry_delay
        id: 28
      - platform: sensor
        friendly_name: alarm.exit_delay
        id: 2
      - platform: sensor
        friendly_name: alarm.siren_duration
        id: 3
      - platform: sensor
        friendly_name: alarm.arm_beep
        id: 4
      - platform: sensor
        friendly_name: alarm.siren
        id: 6
      - platform: sensor
        friendly_name: alarm.arm_sms
        id: 12
      - platform: sensor
        friendly_name: alarm.backlight
        id: 13
      - platform: sensor
        friendly_name: alarm.25
        id: 25
      - platform: sensor
        friendly_name: alarm.message_propel
        id: 27
```


Ciaociao :-)

**Comment by ropmip** [comment]
@agarbato You can test on the services tab (programing tools) the name of the service is localtuya.set_dp
You can set up an alarm in homeassistant with the manual integration
[https://www.home-assistant.io/integrations/manual/](url)
